src/drivers/riscv_plic0.c

```diff
@@ -136,12 +138,14 @@ void __metal_driver_riscv_plic0_init(struct metal_interrupt *controller) {
             /* Initialize ist parent controller, aka cpu_intc. */
             intc->vtable->interrupt_init(intc);
 
-            for (int i = 0; i < num_interrupts; i++) {
```

Comment:
Since we no longer use `num_interrupts`, we should probably remove the variable completely.